src/library.cpp

```diff
@@ -123,6 +125,43 @@ void Library::save()
     m_library.writeBookmarksToFile(kiwix::appendToDirectory(m_libraryDirectory.toStdString(), "library.bookmarks.xml"));
 }
 
+void Library::loadMonitorDir(QString monitorDir)
+{
+    const QDir dir(monitorDir);
+    QStringList newDirEntries = dir.entryList({"*.zim"});
+    QSet<QString> newDir = QSet<QString>(newDirEntries.begin(), newDirEntries.end());
+    QStringList oldDirEntries = KiwixApp::instance()->getSettingsManager()->getMonitorDirZims();
+    QSet<QString> oldDir = QSet<QString>(oldDirEntries.begin(), oldDirEntries.end());
+    QSet<QString> zimSet = newDir - oldDir;
+    QStringList newZims = zimSet.values();
+    KiwixApp::instance()->getSettingsManager()->setMonitorDirZims(newDir.values());
+    auto manipulator = LibraryManipulator(this);
+    auto manager = kiwix::Manager(&manipulator);
+    QString libXMLPath = QString::fromStdString(kiwix::getDataDirectory()) + QDir::separator() + "library.xml";
+    std::vector<std::string> paths{libXMLPath.toStdString()};
+    manager.reload(paths);
+    if (!newZims.empty()) {
+        QString fullBookPath;
+        for (auto book : newZims) {
+            fullBookPath = monitorDir + QDir::separator() + book;
+            if (!manager.addBookFromPath(fullBookPath.toStdString())) {
+                qWarning()<<"Couldn't add "<<fullBookPath<<" in the library.";
+            }
+        }
+    }
```

Comment:
The `if` is unnecessary. Please remove it and declare `fullBookPath` inside the loop.